src/wallet/bdb.cpp

```diff
@@ -591,10 +588,10 @@ void BerkeleyEnvironment::Flush(bool fShutdown)
         return;
     {
         LOCK(cs_db);
-        std::map<std::string, int>::iterator mi = mapFileUseCount.begin();
-        while (mi != mapFileUseCount.end()) {
-            std::string strFile = (*mi).first;
-            int nRefCount = (*mi).second;
+        bool no_dbs_accessed = true;
+        for (auto& db_it : m_databases) {
```

Comment:
I've changed `m_refcount` to track when the database has been flushed by setting it to `-1`. This is being done everywhere `mapFileUseCount.erase()` was happening.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/19335#discussion_r455990656

In commit "walletdb: track database file use as m_refcount within BerkeleyDatabase" (516084a2738364f3922c08d8703fd737541bb2da)

> I've changed `m_refcount` to track when the database has been flushed by setting it to `-1`. This is being done everywhere `mapFileUseCount.erase()` was happening. This allows us to restore the original behavior by checking for `m_refcount < 0`.

Note: I think there is still small change in behavior because m_refcount is initialized to 0 not -1, so never used databases are now treated like used and not yet flushed databases. I don't think the difference actually matters, though.